addons/mass_mailing/models/mailing_trace.py

```diff
@@ -132,6 +132,14 @@ def create(self, vals_list):
                 values['mail_mail_id_int'] = values['mail_mail_id']
         return super().create(vals_list)
 
+    def action_retry(self):
+        traces = self.filtered(lambda t: t.trace_status in ("error", "cancel", "bounce"))
+        if not traces:
+            return
+        retry_mails = traces.mapped('mail_mail_id').ids
+        mail_domain = [('id', 'in', retry_mails)]
```

Comment:
There is a mailing_trace_ids field on mail_mail so you should build your domain with that instead of fetching the mail.mail records, much more efficient.